app/routes/articles.js

```diff
@@ -162,49 +168,56 @@ router.get('/:id', async (req, res) => {
 })
 
 //EDIT Route
-router.get('/:id/edit', checkArticleOwnership, (req, res) => {
-    Article.findById(req.params.id, (err, article) => {
+router.get('/:link/edit', checkArticleOwnership, (req, res) => {
+    if(!req.params.link) return res.redirect('/articles')
+    const encodedLink = encodeURIComponent(req.params.link.replace(SPACES, DASH))
+    Article.findOne({link: encodedLink}, (err, article) => {
         if(err) {
             req.flash('error', 'Oops! Something went wrong!')
             req.log('Article EDIT Route:', err)
             return res.redirect('/')
         }
+        if(!article) {
+            req.flash('error', 'That article does not exist')
+            return res.redirect('/')
+        }
         res.render('pages/article-edit', {title: 'Edit Article', categories: CATEGORIES_LIST, article, method: 'PUT', type: 'edit', limits: ARTICLE_LIMITS})
     })
 })
 
 //UPDATE Route
-router.put('/:id', checkArticleOwnership, (req, res) => {
-    if(req.body.title === undefined) {
+router.put('/:link', checkArticleOwnership, (req, res) => {
+    if(req.params.link === undefined) {
         req.flash('error', 'Oops! Something went wrong!')
         req.log('Article UPDATE Route:', req.body)
         return res.redirect('/')
     }
-    Article.findByIdAndUpdate(req.params.id, {$set: req.body}, err => {
+
+    Article.findOneAndUpdate({link: encodeURIComponent(req.params.link)}, {$set: req.body}, {runValidators: true}, (err, article) => {
         if(err) {
             if(err?.errors?.properties?.type === 'minlength' || err?.errors?.properties?.type === 'maxlength') {
                 return res.render('error', {code: '401', msg: 'Invalid input length.'})
             }
 
             if(err._message === 'Article validation failed') {
                 req.flash('error', 'Invalid input lengths, please try again.')
-                return res.redirect(`/articles/${req.params.id}/edit`)
+                return res.redirect(`back`)
             }
             
             req.flash('error', 'Oops! Something went wrong!')
             req.log('Article UPDATE Route:', err)
             return res.redirect('/')
         }
-
+        if(article.title !== req.body.title) article.link = encodeURIComponent(req.body.title.replace(SPACES, DASH))
         req.flash('success', 'Successfully updated your article!')
-        res.redirect('/articles/' + req.params.id)
+        res.redirect('/articles/' + encodeURIComponent(req.params.link))
```

Comment:
if you were able to navigate to this route through `req.params.link`, then you can navigate back as well, there's no need for encoding